python/cudf/cudf/core/frame.py

```diff
@@ -236,9 +237,21 @@ def _concat(cls, objs, axis=0, ignore_index=False, sort=False):
                     result_index_length += len(obj)
                     empty_has_index = empty_has_index or len(obj) > 0
 
-        # Get a list of the unique table column names
-        names = [name for f in objs for name in f._column_names]
-        names = OrderedDict.fromkeys(names).keys()
+        if join == "inner":
+            index = objs[0].columns if len(objs) > 0 else pd.Index([])
+            for other in objs[1:]:
+                index = index.intersection(other.columns)
+            names = list(index)
+            for obj in objs:
+                columns_to_drop = [
+                    col for col in obj.columns if col not in names
+                ]
+            obj = obj.drop(columns=columns_to_drop)
+
+        else:
+            # Get a list of the unique table column names
+            names = [name for f in objs for name in f._column_names]
+            names = OrderedDict.fromkeys(names).keys()
```

Comment:
```suggestion
        elif join == "outer":
            # Get a list of the unique table column names
            names = [name for f in objs for name in f._column_names]
            names = OrderedDict.fromkeys(names).keys()
        else:
            raise ValueError(
                "Only can inner (intersect) or outer (union) join the other axis"
            )
```

Better to have a clear error on what we support instead of having any value run into `outer` block.

Reply:
Sounds good!

Reply:
Added these changes. I think it looks good to go @galipremsagar